### PR TITLE
Fix for issue #2832

### LINKED ISSUE #2832: Delete app template which has versions pops error
**Describe the Bug**
Delete app template with existing versions will cause a Internal Server Error

image

**Versions Used**
KubeSphere: v3.0.0

**Expected behavior**
It's a business logical error which could be handled and prompted more friendly, no need to cause an internal server error.

### PR DESCRIPTION
**Pull request by Howie59**
Signed-off-by: Howie66 <haoweili35@gmail.com>

**What type of PR is this?**
> /kind bug

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #2832 #2860

### CODE REVIEW

File: pkg/kapis/openpitrix/v1/handler.go [hunk 1]
```diff
@@ -927,6 +927,11 @@ func handleOpenpitrixError(resp *restful.Response, err error) {
 		api.HandleBadRequest(resp, nil, err)
 		return
 	}
+	if status.Code(err) == codes.FailedPrecondition {
```
Comment (zryfish) [inline pkg/kapis/openpitrix/v1/handler.go:930]: to make it simpler, could squash this and the upper if into one statement
```suggestion
	if status.Code(err) == codes.InvalidArgument || status.Code(err) == codes.FailedPrecondition {
```

**Comment by zryfish** [comment]
@Howie59  There is one more thing to do, please sign the commit with git commit -s

**Comment by Howie59** [comment]
> @Howie59 There is one more thing to do, please sign the commit with git commit -s

i pushed again, the link is https://github.com/kubesphere/kubesphere/pull/3003

**Pull request closed without merging**